lib/int3.h

```diff
@@ -130,29 +123,29 @@ class int3
 	}
 
 	//returns squared distance on Oxy plane (z coord is not used)
-	ui32 dist2dSQ(const int3 & o) const
+	constexpr ui32 dist2dSQ(const int3 & o) const
 	{
 		const si32 dx = (x - o.x);
 		const si32 dy = (y - o.y);
 		return (ui32)(dx*dx) + (ui32)(dy*dy);
 	}
 	//returns distance on Oxy plane (z coord is not used)
-	double dist2d(const int3 & o) const
+	constexpr double dist2d(const int3 & o) const
 	{
-		return std::sqrt((double)dist2dSQ(o));
+		return vstd::ct_sqrt((double)dist2dSQ(o));

```

Comment:
Is there any need for compile-tile distance calculation?
This code will also execute in runtime. Meaning potential change in behavior or performance.

Especially since old distance computation had double as result but now it uses int.

Reply:
Made dist2d as non-constexpr, as you request.